pkg/reconciler/v1alpha1/route/reconcile_resources.go

```diff
@@ -131,3 +137,59 @@ func (c *Reconciler) updateStatus(ctx context.Context, route *v1alpha1.Route) (*
 	c.Recorder.Eventf(route, corev1.EventTypeNormal, "Updated", "Updated status for route %q", route.Name)
 	return updated, nil
 }
+
+// Update the lastPinned annotation on revisions we target so they don't get GC'd.
+func (c *Reconciler) reconcileTargetRevisions(ctx context.Context, t *traffic.TrafficConfig, route *v1alpha1.Route) error {
+	rev_errs := []error{}
+	for _, target := range t.Targets {
+		for _, rt := range target {
+			tt := rt.TrafficTarget
+			rev, err := c.revisionLister.Revisions(route.Namespace).Get(tt.RevisionName)
+			if apierrs.IsNotFound(err) {
+				c.Logger.Infof("Unable to update lastPinned for missing revision %q", tt.RevisionName)
+				continue
+			} else if err != nil {
+				rev_errs = append(rev_errs, err)
+				continue
+			}
+
+			// Enforce a delay before performing an update on lastPinned to avoid excess churn
+			if rev.Annotations == nil {
+				rev.Annotations = make(map[string]string)
+			}
+
+			lastPin, err := rev.GetLastPinned()
+			if err != nil && err.(v1alpha1.LastPinnedParseError).Type == v1alpha1.AnnotationParseErrorTypeInvalid {
+				rev_errs = append(rev_errs, err)
+				continue
+			}
+
+			if lastPin.Add(1 * time.Minute).After(c.clock.Now()) {
+				// Skip updating if were not to the decay limit
```

Comment:
is 1 minute the decay?  Can we make this configurable and validate it against the other values?

Reply:
Yep, I had to share the gc config between configuration and route reconcilers in order to do this, not sure if there's a better way...